index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
I simply am not clear on what constitutes invalid operations in this case.

Is `Maybe(42)` valid?  `Maybe('foo')`?  Would it be all right that `Maybe(42).concat(Maybe('foo'))` throws an exception since 7 doesn't have a `concat` method?  Would you be willing to live with whatever weirdness derived from `Maybe('foo').concat(Maybe([1, 2, 3]))` due to type conversion?  (`Maybe('foo1,2,3')`?)

Those are the concerns I see, in any case.